js/foam/u2/md/WithToolbar.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.u2.md',
+  name: 'WithToolbar',
+  extends: 'foam.u2.Element',
+  requires: [
+    'foam.u2.md.Toolbar',
+    'foam.u2.md.ToolbarAction',
+  ],
+
+  imports: [
+    'stack',
+  ],
+
+  properties: [
+    {
+      name: 'title',
+    },
+    {
+      name: 'body_',
+      documentation: 'The body element. add() calls are forwarded here.',
+      factory: function() {
+        return this.Y.E('div').cls(this.myCls('body'));
+      }
+    },
+    {
+      name: 'pendingChildren_',
+      factory: function() {
+        return [];
+      }
+    },
+    {
+      name: 'backAction_',
+      factory: function() {
+        return this.ToolbarAction.create({ data: this, action: this.BACK });
+      }
+    },
+    {
+      name: 'toolbar_',
+      factory: function() {
+        var t = this.Toolbar.create({ title$: this.title$ });
+        t.addLeftActions([this.backAction_]);
+        return t;
+      }
+    },
+    ['addPassthrough_', false]
```

Comment:
It's needed for the hack of adding things to a child view. We need to discuss this further today, it's an important use case that's going to keep coming up.